node/engine/engine.go

```diff
@@ -227,7 +222,7 @@ func (e *Engine) run(ctx context.Context) {
 				e.logger.Printf("Objective %s is complete & returned to API", obj.Id())
 				e.metrics.RecordObjectiveCompleted(obj.Id())
 			}
-			e.toApi <- res
+			e.eventHandler(res)
```

Comment:
I think this means that the engine `run` loop now blocks on the event handler, which means there might be a slight performance hit with this change? `handleEngineEvent` doesn't do too much, so it's quite possible that performance hit is negligible. 

It's too bad we no longer have the testground test, so we could see if there was any noticeable performance difference.

Reply:
Yes too bad we don't have testground. Maybe we can measure performance in another way for this change. 

My hunch is that there will not be a negative effect on performance. 

Take the following thought experiment: take as much synchronous code as possible in the engine and run it in goroutines. Would that improve performance? I think only if something were blocking on I/O or network. Or (maybe) heavy crypto operations, but then we would likely need to be using multiple CPUs to notice a significant difference. If putting very fast operations in a goroutine won't improve performance, removing the goroutine as this PR does shouldn't have a negative impact.

Reply:
Quick and dirty performance test:
```
~/s/go-nitro ❯❯❯ go test ./node_test -run TestComplexIntegrationScenario                                                                                                                                         ✘ 1 main ✭
ok  	github.com/statechannels/go-nitro/node_test	0.665s
~/s/go-nitro ❯❯❯ go test ./node_test -run TestComplexIntegrationScenario                                                                                                                                             main ✭
ok  	github.com/statechannels/go-nitro/node_test	0.506s
~/s/go-nitro ❯❯❯ go test ./node_test -run TestComplexIntegrationScenario                                                                                                                                             main ✭
ok  	github.com/statechannels/go-nitro/node_test	0.508s
~/s/go-nitro ❯❯❯ go test ./node_test -run TestComplexIntegrationScenario                                                                                                                                             main ✭
ok  	github.com/statechannels/go-nitro/node_test	0.500s
~/s/go-nitro ❯❯❯ go test ./node_test -run TestComplexIntegrationScenario                                                                                                                                             main ✭
ok  	github.com/statechannels/go-nitro/node_test	0.513s
                                                                                                                                                     ✘ 1 main ✭
```
```
~/s/go-nitro ❯❯❯ git checkout engine-event-handler          
Switched to branch 'engine-event-handler'
Your branch is up to date with 'origin/engine-event-handler'.
~/s/go-nitro ❯❯❯ go test ./node_test -run TestComplexIntegrationScenario                                                                                                                             engine-event-handler ✭
ok  	github.com/statechannels/go-nitro/node_test	0.581s
~/s/go-nitro ❯❯❯ go test ./node_test -run TestComplexIntegrationScenario                                                                                                                             engine-event-handler ✭
ok  	github.com/statechannels/go-nitro/node_test	0.496s
~/s/go-nitro ❯❯❯ go test ./node_test -run TestComplexIntegrationScenario                                                                                                                             engine-event-handler ✭
ok  	github.com/statechannels/go-nitro/node_test	0.496s
~/s/go-nitro ❯❯❯ go test ./node_test -run TestComplexIntegrationScenario                                                                                                                             engine-event-handler ✭
ok  	github.com/statechannels/go-nitro/node_test	0.506s
~/s/go-nitro ❯❯❯ go test ./node_test -run TestComplexIntegrationScenario                                                                                                                             engine-event-handler ✭
ok  	github.com/statechannels/go-nitro/node_test	0.488s
~/s/go-nitro ❯❯❯
```

Reply:
> I might be nitpicking but I don't think you need multiple CPUs, just multiple cores, which is pretty common. I think it would be rare to encounter a cpu with only a single core these days.

Yes I meant multiple processors / cores / logical CPUs. You're right that most machines have multiple cores and it looks like these can all be used by default since Go 1.15 https://medium.com/a-journey-with-go/go-gomaxprocs-live-updates-407ad08624e1.